File: Daily Test/C/Integer - Last Set Bit 31.8.2025.c
#include <stdio.h>
#include <math.h>

int main() {
    int N;
    scanf("%d", &N);
    int val[N];
    for (int i = 0; i < N; i++) scanf("%d", &val[i]);
    
    long long tot = 0;
    for (int i = 0; i < N; i++) {
        int num = val[i], l = 0;
        while ((num & 1) == 0) {
            l++;
            num >>= 1;
        }
        tot += (long long)pow(2, l);
    }
    printf("%lld\n", tot);
    return 0;
}

File: Daily Test/Java/Integer - Last Set Bit 31.8.2025.java
import java.util.*;
public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int N = sc.nextInt();
        int[] val = new int[N];
        for (int i = 0; i < N; i++) val[i] = sc.nextInt();

        long tot = 0;
        for (int x : val) {
            int l = 0;
            while ((x & 1) == 0) {
                l++;
                x >>= 1;
            }
            tot += (long)(Math.pow(2, l));
        }
        System.out.println(tot);
        sc.close();
    }
}